services/management/backup/backups_service.go

```diff
@@ -0,0 +1,148 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package backup
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes"
+	backupv1beta1 "github.com/percona/pmm/api/managementpb/backup"
+	"github.com/pkg/errors"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// BackupsService represents backups API.
+type BackupsService struct {
+	db *reform.DB
+}
+
+// NewBackupsService creates new backup API service.
+func NewBackupsService(db *reform.DB) *BackupsService {
+	return &BackupsService{
+		db: db,
+	}
+}
+
+// ListBackups returns a list of all backups.
+func (s *BackupsService) ListBackups(context.Context, *backupv1beta1.ListBackupsRequest) (*backupv1beta1.ListBackupsResponse, error) {
+	q := s.db.Querier
+
+	backups, err := models.FindBackups(q)
+	if err != nil {
+		return nil, err
+	}
+
+	locationIDs := make([]string, 0, len(backups))
+	for _, b := range backups {
+		locationIDs = append(locationIDs, b.LocationID)
+	}
+	locations, err := models.FindBackupLocationsByIDs(q, locationIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	serviceIDs := make([]string, 0, len(backups))
+	for _, b := range backups {
+		serviceIDs = append(serviceIDs, b.ServiceID)
+	}
+
+	services, err := models.FindServicesByIDs(q, serviceIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	backupsResponse := make([]*backupv1beta1.Backup, 0, len(backups))
+	for _, b := range backups {
+		convertedBackup, err := convertBackup(services, locations, b)
+		if err != nil {
+			return nil, err
+		}
+		backupsResponse = append(backupsResponse, convertedBackup)
+	}
+	return &backupv1beta1.ListBackupsResponse{
+		Backups: backupsResponse,
+	}, nil
+}
+
+func convertBackup(
```

Comment:
just detail, personal opinion. Backup should be as first argument, we are converting backup